Rework retail environment setup and formulas pages (#877)
SUSE/doc-susemanager: modules/retail/pages/retail-install-setup.adoc

[[retail-install-setup]]
= Setting up

> This section requires installed and configured {susemgr} Server,at least one {smr} branch server and at least one {susemgr} build host.

Reviewer: ```suggestion
This section requires an installed and configured {susemgr} Server, one or more {smr} branch servers, and one or more {susemgr} build host.
```

Next steps in configuring {smr} environment are:

. Prepare POS images
. Configure {smr} branch server services
. Synchronize POS images to {smr} branch servers

> These steps are usually performed multiple times in lifetime of {smr} and are loosely related.

Reviewer 2: What is this sentence trying to say? "You'll do this a lot, get used to it?"

Author: In the typical setup customer have only one or small number of build hosts and images, so this step (POS Images) is done only few times. But then have tens or hundreds, even thousands of branch servers and these needs to be setup all the time. 

I structured the doc that after suma is configured, build host is configured and image is build and then numerous branches are configured and then images are synchronized.

But for the first branch, you need to have images to synchronize. For the second branch, you do not need new image. You can reuse the existing one so you skip POS Images step.
When you will build new image and you don't have to do any reinitialization of branches, just synchronize them, so you skip branch services configuration step.

There is some relation between images and branches, but it is not that one require the other, but rather that one needs other for working system and both parts are changing during the product lifetime.

Now how do we write this into the documentation in understandable way?

Reviewer 2: For starters, it's going to need more than three sentences, I think!

Reviewer 2: I can't check out your branch to edit directly, but I would do something like this (from line 4):

`
To set up an {smr} environment, you will need to have already installed and configured {susemgr} Server, have one or more {smr} branch server, and one or more {susemgr} build host.

This section covers how to configure your {smr} environment, including:
. Prepare POS images
￼. Configure services on the branch server
. Synchronize POS images to the branch servers

The very first time you set up an {smr} environment, you will need to perform all three steps.
You will need to revisit some of these steps later on as you are working with {smr}.

For example, the first time you configure the branch server, you will need to have images prepared for synchronization.
If you are configuring more than one branch server, you will only need to synchronize images for the first branch server.
Subsequent branch servers can use the same image.

If you have an existing environment, and need to build new images, you do not need to re-initialize the branches.
You will need to synchronize the images, and can skip setting up the services on the branch server.
`

> Typically POS images are periodically rebuild when updated packages are available and then synchronized to the {smr} branch servers before update window.

Reviewer 2: ```suggestion
Usually, POS images are rebuild when updated packages are available, and synchronized to the branch servers before the update window opens.
```

> Configuration of {smr} branch server services has to be done each time new branch server is registered.

Reviewer: ```suggestion
Configuration of {smr} branch server services have to be done each time a new branch server is registered.
```

> Configuration of {smr} branch server services have to be done each time a new branch server is registered.

Reviewer 2: ```suggestion
You will need to configure the services on the branch server every time a new branch server is registered.
```

== Prepare and Build Terminal Images

For information about {susemgr} image building, see xref:administration:image-management.adoc[].

{smr} POS images are images specifically tailored for {smr} environment and designed to be deployed using PXE booting mechanism.

> === POS image templates

Reviewer: ```suggestion
=== POS Image Templates
```

As starting point, {suse} provides basic templates at https://github.com/SUSE/manager-build-profiles/tree/master/OSImage[].

> These templates needs to be adapted for specific usecases, for example by including specific applications, configuration and users.

Reviewer: ```suggestion
These templates need to be adapted for specific usecases, for example by including specific applications, configuration settings, and users.
```

[IMPORTANT]
====

> POS templates do not describe any system user. Using {suse} provided templates as is will result in system without possibility of user login. Salt management of the system is unaffected, however.

Reviewer 2: ```suggestion
By default, POS templates do not include a system user. 
You will not be able to login as a user to a system that has been installed with a {suse} provided template. 
This does not apply to Salt-managed terminals.
```

This admon needs a remedy.

Author: > This does not apply to Salt-managed terminals.

This is not correct. What I meant to say is that the salt management of the terminals is still possible and it is unrelated to missing user configuration. In other words, it is also possible to add user using salt after the terminal was deployed.

Reviewer 2: Better now?

====

> === Terminals Based on SLES{nbsp}11 SP{nbsp}3

Reviewer 2: ```suggestion
=== SLES{nbsp}11 SP{nbsp}3 Terminals
```

POS Terminals based on {sles}{nbsp}11 SP{nbsp}3 can be deployed in much the same way as other terminals, with a few differences.

* You must use the SLES{nbsp}11 template
* SLES{nbsp}11 images need to be activated with the [systemitem]``SLES11 SP3 i586`` and [systemitem]``SLEPOS 11 SP3 i586`` channels

[IMPORTANT]
====
Ensure that SLES{nbsp}11 images are built on the SLES{nbsp}11 build host.
Building on the incorrect build host will cause your build to fail.
====

[WARNING]
====
If you are building images for SLES{nbsp}11 using profiles from an HTTPS git repository that uses TLS 1.0 or greater, it will fail.
SLES{nbsp}11 does not support later versions of TLS.
You will need to clone the repository locally in order to use it for building.
====

> == Configure {smr} Branch Server services

Reviewer 2: ```suggestion
== Configure Services on the Branch Server
```

Before you configure the branch server, ensure you have decided on networking topology, and know the minion ID of the branch server.
For the information about the possible network topologies, see xref:retail:retail-network-arch.adoc[].

> All configuration of branch server services is done on {susemgr} server.

Reviewer 2: ```suggestion
Configure branch server services from the {susemgr} Server.
```

> The configuration is then applied to the selected branch server through salt states.

Reviewer 2: ```suggestion
The configuration is then applied to the selected branch server using Salt states.
```

> {susemgr} Formulas with Forms functionality is used to configure branch server services, however there are multiple ways to configure them.

Reviewer: ```suggestion
{susemgr} Formulas with Forms functionality is used to configure branch server services, however there are multiple ways to configure them:
```

. Apply highstate on the branch server.
You can do this through the {webui}, or by running this command:
+
----
salt <branch_server_minion_id> state.apply
----

Similar results can be achieved by using mass import command line tool.

.Procedure: Configuring Branch Server Formulas With a Mass Import Tool

. Prepare branch specific YAML file:
+
For example, create branch.yaml file with content:
+
----
branches:
  <branch_server_minion_id>:
    branch_prefix: branch1
    server_name: branchserver1
    server_domain: example.com
    nic: eth1
    dedicated_nic: true
    configure_firewall: true
    branch_ip: 192.168.2.1
    netmask: 255.255.255.0
    dyn_range:
        - 192.168.2.10
        - 192.168.2.250
----
+
For more information about mass import tool, see xref:retail:retail-mass-config.adoc[].
. Import branch information from YAML file to {susemgr}
+
----
retail_yaml --from-yaml branch.yaml
----
. Verify that your changes have been configured correctly by checking the {susemgr} {webui} branch server system formulas.
. Apply highstate on the branch server.

[WARNING]
====

> Both [command]``retail_branch_init`` and [command]``retail_yaml`` commands overwrites existing configuration of specified branch server.

Reviewer: ```suggestion
Both [command]``retail_branch_init`` and [command]``retail_yaml`` commands override existing configuration settings of the specified branch server.
```

Author: I don't think override is the correct verd here. That command literally changes the stored configuration

Reviewer: I edited the suggestion: changing back to overwrite, but using plural form.

(I'm frankly admit that I treat both verbs as synonyms and usually preferring override. But I can imaging that there are differences.)

Reviewer 2: Override has a lot more usage in the English corpus, because it has a greater variety of uses (https://books.google.com/ngrams/graph?content=override%2C+overwrite&year_start=1800&year_end=2000&corpus=15&smoothing=3&share=&direct_url=t1%3B%2Coverride%3B%2Cc0%3B.t1%3B%2Coverwrite%3B%2Cc0). AHD has 5 definitions for override as a transitive verb, and 4 as a noun, while overwrite only lists two, because it's really specific to writing, rather than a general sense of going over. In most software contexts, override is more accurate, because we're not just talking about putting new words over old words, but changing settings, which is a greater effect.

Author: I don't know, at the end you are the experts here :)
I just put 'override vs overwrite' in the google and the first link for me was https://english.stackexchange.com/questions/88400/when-to-use-override-and-overwrite where they states:

> "Override" is the cancellation of some previous action or decision. "Overwrite" specifically refers to something being written over something previously written. In other words, actions and decisions are overriden. Information is overwritten. Furthermore it is overwritten with new information.

Which is exactly what I'm pointing out. Both `retail_branch_init` and `retail_yaml` destroys previous configuration and write completely new one. Which for me sounds like _"something being written over something previously written"_

Reviewer 2: Yes, but I'm thinking more about the effect of that: it's not the words in the configuration file so much as the settings you are changing. Consider: if you change the settings in the UI (by clicking a button, or selecting from a list), that's overriding, not overwriting. So if you do the same action in a configuration file, it could technically be either. In which case, we should pick the word that applies more generally, which is overriding.

Author: Picked the `override` variant :+1:

====

After the initial configuration done by command line tools, branch server configuration can be further adjusted in {susemgr} {webui} through branch server formulas.

> === Create Required System Groups

Reviewer 2: This type of heading pre-supposes a procedure, but there isn't one ...

Author: Right, I'll add a procedure header below.

Author: At the end I changed the header to `Required System Groups`. I couldn't figure out what procedure to add as I am xrefing to the other part of doc for howto.

For more information about hardware type groups, see xref:retail:retail-deploy-terminals.adoc[].

Branch server groups are named after branch server prefixes, for example group name [systemitem]``B0001`` for branch server prefix [systemitem]``BOO1``.

You can create system groups using the {susemgr} {webui}.
Navigate to menu:Systems[System Groups] and click btn:[Create System Group].

For more information about system groups, see xref:reference:systems/system-groups.adoc[].

[NOTE]
====
{smr} command line tools create required system groups and branch group automatically.
====

== Synchronize Images to the Branch Server

The OS image you use on the {susemgr} server must be synchronized for use to the branch server.
You can do this with the Salt [command]``image-sync`` state, part of the [systemitem]``Image Synchronization Formula``.

.Procedure: Synchronizing Images to the Branch Server

. On the {susemgr} server, run this command:
+
----
salt <branch_server_minion_id> state.apply image-sync
----
. The image details will be transferred to [path]``/srv/saltboot`` on the branch server.

> Synchronization can also be enabled to be done automatically on applying highstate on {smr} branch server using [systemitem]``Image Synchronization Formula``.

Reviewer 2: ```suggestion
You can also set synchronization to run automatically on the branch server.
Configure the image synchronization formula to apply the highstate regularly.
```
